packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.ts

```diff
@@ -392,7 +389,11 @@ export default ({ api, coreSagas, networks }) => {
       }
       yield call(fetchBalances)
       yield call(saveGoals, firstLogin)
-      yield put(actions.goals.runGoals())
+      // We run goals in accountResetSaga in this case
```

Comment:
expand comment - if it's account reset, we need to wait until user's metadata is written until we run the goals saga, otherwise it will run before /users/current returns. That call let's us know if we should show the 'Resubmit KYC' docs modal